src/table/cells/handle-key-press.js

```diff
@@ -27,12 +27,32 @@ export const arrowKeysNavigation = (evt, rowAndColumnCount, cellCoord, selState)
     case 'ArrowUp':
       nextRow > 0 && (!isSelectedTable || nextRow !== 1) && nextRow--;
       break;
-    case 'ArrowRight':
-      nextCol < rowAndColumnCount.columnCount - 1 && !isSelectedTable && nextCol++;
+    case 'ArrowRight': {
+      if (nextCol < rowAndColumnCount.columnCount - 1) !isSelectedTable && nextCol++;
+      else if (!isSelectedTable) {
+        // move to the next row
+        // if we are at the end of the table do nothing
+        if (nextRow + 1 > rowAndColumnCount.rowCount - 1) break;
+        else {
+          nextRow = nextRow + 1;
+          nextCol = 0;
+        }
```

Comment:
The logic here is quite confusing. First off, you are mixing `if` and `&&` outside the if. On row 31, make it into one expression inside the if and add {} to wrap `nextCol++`. Second, I think the second case can be reduced to one if statement:
`!isSelectableTable && nextRow + 1 < rowAndColumnCount.rowCount - 1` (not sure if you need the + 1)

You can also use `nextRow++` on row 37 for consistency

Same thing goes for the `arrowLeft` case

Reply:
Changes Done!